app/src/input_manager.c

```diff
@@ -741,14 +741,26 @@ sc_input_manager_process_mouse_wheel(struct sc_input_manager *im,
     int mouse_y;
     uint32_t buttons = SDL_GetMouseState(&mouse_x, &mouse_y);
 
+    float hscroll;
+    float vscroll;
+    if (SDL_VERSION_ATLEAST(2, 0, 18)) {
+      // right is positive
+      hscroll = CLAMP(-event->preciseX, -1.0f, 1.0f);
+      // up is positive
+      vscroll = CLAMP(event->preciseY, -1.0f, 1.0f);
+    } else {
+      hscroll = CLAMP(-event->x, -1, 1);
```

Comment:
Nowadays the horizontal scrolling is actually inverted, so negated `x` and `preciseX`.

Reply:
Hmmm, looks suspicious. See https://github.com/Genymobile/scrcpy/issues/966#issuecomment-560394359

Reply:
This also happens to mouses with horizontal scrolling wheels. I will test on a mac tomorrow.

Reply:
Confirmed it's also inverted on macOS when using a mouse.

Reply:
What's the value of `event->direction` in each case?

Concretely, could you list for each case (mouse/touchpad on windows/macOS):
 - `event->direction`
 - `event->x`
 - `event->y`
 - "inverted" or "not inverted"

cc @fsw0422 as the author of #966 (I do not want to break the fix for #966)

Reply:
I won't be with my mac until Friday, will try that later.